### PR TITLE
chore(integer): addition test based on trivial inputs

### PR DESCRIPTION
**Pull request by tmontaigu**
This adds `overflowing_add` and `add` tests that
are on trivial inputs. As these are faster to run they can be more extensive than on true encryptions

This also binds the advanced_add_assign functions tests to include overflow computation

On a standard laptop with 1 test thread it takes ~7 minutes to run these trivial tests

### CODE REVIEW

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs [hunk 1]
```diff
@@ -69,24 +90,67 @@ where
...
+        let mut result = lhs.clone();
+        let overflowed = sks
+            .advanced_add_assign_with_carry_at_least_4_bits(
+                &mut result.blocks,
+                &rhs.blocks,
+                None,
+                OutputFlag::Carry,
+            )
+            .unwrap();
+        (result, overflowed)
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs:150]: no carry management here ? I see it done in other test functions

**Review by IceTDrinker — COMMENTED** [review]
small question otherwise looks good

File: tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs]: Aded them just for future proofing if we later change the extensive_trivial test to also have non clean inputs

**Review by IceTDrinker — APPROVED** [review]
🚀
